### PR TITLE
fix: json parse error on Sources component and TypeError in Mapping component

### PR DESCRIPTION
**Pull request by ghost**
* Clicking on a source sometimes does not work, the page has to be refreshed -> JSON parse error in Sources component
* Refreshing on a ressource page will lead to a bug (dark page) -> TypeError in Mapping component

### CODE REVIEW

File: client/src/components/sources/index.tsx [hunk 1]
```diff
@@ -66,14 +61,25 @@ const SourcesView = (): React.ReactElement => {
...
   const onSelectSource = async (source: ISelectedSource) => {
-    if (source.credential)
-      source.credential.schema = JSON.parse(source.credential.schema as string);
+    try {
+      if (source.credential)
+        source.credential.schema = JSON.parse(
+          source.credential.schema as string
+        );
+    } catch {
+      // No parsing needed as `schema` attribute is already a valid js object
```
Comment (Jasopaum) [inline client/src/components/sources/index.tsx:81]: Do you when it's a string and not an object?

File: client/src/components/sources/sourceCard.tsx [hunk 2]
```diff
@@ -88,6 +88,6 @@ export const SourceCard = ({
         onClose={() => setIsCollaboratorDialogOpen(false)}
         onUpdate={onUpdate}
       />
-    </React.Fragment>
+    </>
```
Comment (Jasopaum) [inline client/src/components/sources/sourceCard.tsx:91]: Why did you remove that? I don't think I have any preference but this and `React.Fragment` are equivalent

File: client/src/components/sources/index.tsx [hunk 3]
```diff
@@ -66,14 +61,19 @@ const SourcesView = (): React.ReactElement => {
...
+
   const [deleteSource, { loading: deletingSource }] = useMutation(
     mDeleteSource,
     { update: removeSourceFromCache, onError: onError(toaster) }
   );
 
   const onSelectSource = async (source: ISelectedSource) => {
-    if (source.credential)
-      source.credential.schema = JSON.parse(source.credential.schema as string);
+    if (source.credential && typeof source.credential.schema === 'string')
```
Comment (simonvadee) [inline client/src/components/sources/index.tsx:75]: I thought the JSON.parse error was thrown when source.credential.schema is an empty string, isn't it ?
I think you can just do
```javascript
if (source.credential?.schema)
...
```
no ?

Comment (simonvadee) [reply client/src/components/sources/index.tsx:75]: ok I misunderstood the issue, let's put this in the reducer

**Review by simonvadee — APPROVED** [review]
